feat/src/main/java/com/example/feat/course_detail/controller/CourseDetailViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.feat.course_detail.controller
+
+import androidx.lifecycle.viewModelScope
+import com.example.core.presentation.StateAndEventViewModel
+import com.example.domain.usecase.course_detail.CourseDetailUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class CourseDetailViewModel @Inject constructor(
+    private val courseDetailUseCase: CourseDetailUseCase
+) : StateAndEventViewModel<CourseDetailUiState, CourseDetailUiEvent>(initialState = CourseDetailUiState()) {
+    override suspend fun handleEvent(event: CourseDetailUiEvent) {
+        when (event) {
+            is CourseDetailUiEvent.FetchCourseDetail -> fetchCourseDetail(event.id)
+        }
+    }
+
+    private suspend fun fetchCourseDetail(id: String) {
+        setUiState { copy(isLoading = true) }
+        viewModelScope.launch {
+            courseDetailUseCase.fetchCourseDetail(id).collect { either ->
+                if (either.isRight()) {
+                    setUiState { copy(courseDetail = either.rightValue()) }
```

Comment:
Write an additional state to handle the failed data, please.

Reply:
here i want either the result is right or left => `setUiState { copy(isLoading = false) }`